renderdoc/replay/replay_driver.cpp

```diff
@@ -565,7 +565,7 @@ bool HighlightCache::FetchHighlightPositions(const MeshDisplay &cfg, FloatVector
   {
     uint32_t dim = PatchList_Count(cfg.position.topo);
 
-    uint32_t v0 = uint32_t(idx / dim) * dim;
+    uint32_t v0 = dim == 0 ? 0 : (uint32_t(idx / dim) * dim);
```

Comment:
`PatchList_Count` is just calculating the number of control points for a given enum value, since the patchlist entries are all sequential one after another starting from `PatchList_1CPs`. That means subtracting the input enum from `PatchList_1CPs` gives you `0, 1, 2, 3, 4...` etc. So adding 1 to that value then gives you back the number of control points.

The problem with the latest revision is you're adding 1 to the whole expression - that means that the `0` return value for invalid inputs will become `1` instead. I think a better fix would be like this:

```cpp
  return uint32_t(topology) < uint32_t(Topology::PatchList_1CPs)
             ? 0
             : uint32_t(topology) - uint32_t(Topology::PatchList_1CPs) + 1;
```

So the increment only happens in the case where we're doing the subtraction.